cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,128 @@
-        Metadata:
-            test_flag: rest
-        """
-        rest_vm = appliance.rest_api.collections.vms.get(name=vm)
-        event = dict(
-            status=fauxfactory.gen_alphanumeric(),
-            message=fauxfactory.gen_alphanumeric(),
-            event=fauxfactory.gen_alphanumeric(),
-        )
-        if from_detail:
-            responses = [rest_vm.action.add_lifecycle_event(**event)]
+    def __init__(self, vm, event):
+        self.vm = vm
+        self.action = event
+
+        if self.action == 'create':
+            self.tl_event = 'VmDeployedEvent'
+            self.tl_category = 'Creation/Addition'
+            self.db_event = ''
+            self.emit_cmd = self._setup_vm()
+        elif self.action in {'start', 'resume'}:
+            self.tl_event = 'VmPoweredOnEvent'
+            self.tl_category = 'Power Activity'
+            self.db_event = ''
+            self.emit_cmd = self._power_on()
+        elif self.action == 'stop':
+            self.tl_event = 'VmPoweredOffEvent'
+            self.tl_category = 'Power Activity'
+            self.db_event = ''
+            self.emit_cmd = self._power_off()
+        elif self.action == 'suspend':
+            self.tl_event = 'VmSuspendedEvent'
+            self.tl_category = 'Power Activity'
+            self.db_event = ''
+            self.emit_cmd = self._suspend()
+        elif self.action == 'rename':
+            self.tl_event = 'VmReconfiguredEvent'
+            self.tl_category = 'Configuration/Reconfiguration'
+            self.db_event = ''
+            self.emit_cmd = self._rename_vm()
+        elif self.action == 'storage':
+            self.tl_event = 'VmReconfiguredEvent'
+            self.tl_category = 'Storage'
+            self.db_event = ''
+            self.emit_cmd = self._rename_vm()
         else:
-            responses = appliance.rest_api.collections.vms.action.add_lifecycle_event(
-                rest_vm, **event)
-        assert appliance.rest_api.response.status_code == 200
-        for response in responses:
-            assert response["success"] is True, "Could not add event"
-
-        # DB check
-        lifecycle_events = appliance.db.client["lifecycle_events"]
-        events_list = list(appliance.db.client.session.query(lifecycle_events).filter(
-            lifecycle_events.message == event["message"],
-            lifecycle_events.status == event["status"],
-            lifecycle_events.event == event["event"],
-        ))
-        assert events_list, "Could not find the lifecycle event in the database"
-
-
-@pytest.mark.manual
-def test_policy_events():
-    pass
-
-
-@pytest.mark.manual
-def test_timelines_ui():
-    pass
+            assert False, logger.warn('VMEvent did not get the proper parameter.')
+
+    def _setup_vm(self):
+        logger.info("Will set up the VM on the provider")
+        self.vm.create_on_provider()
+        return self.vm.provider.mgmt.does_vm_exist(self.vm.name)
```

Comment:
OPTIONAL: I suspect create_on_provider already checks this. there may exist some additional option to enable such check.